src/index/CompressedRelation.h

```diff
@@ -449,14 +456,12 @@ class CompressedRelationReader {
   IdTableGenerator asyncParallelBlockGenerator(
       auto beginBlock, auto endBlock, ad_utility::File& file,
       std::optional<std::vector<size_t>> columnIndices,
-      TimeoutTimer timer) const;
+      std::shared_ptr<ad_utility::AbortionHandle> abortionHandle) const;
 
   // A helper function to abstract away the timeout check:
-  static void checkTimeout(
-      const ad_utility::SharedConcurrentTimeoutTimer& timer) {
-    if (timer) {
-      timer->wlock()->checkTimeoutAndThrow("IndexScan :");
-    }
+  static void checkAbortion(
+      const std::shared_ptr<ad_utility::AbortionHandle>& abortionHandle) {
+    abortionHandle->throwIfAborted("IndexScan"sv);
```

Comment:
If you do it like this, 
you should make sure that the `shared_ptr` is not null.
At least use an `AD_EXPENSIVE_CHECK`, but maybe you can also check everytime.